app/services/jobs/return-logs/generate-return-logs.service.js

```diff
@@ -0,0 +1,161 @@
+
+  return results
+}
+
+async function _metadata (summer, endDate, requirements) {
+  return {
+    description: requirements.siteDescription,
+    isCurrent: requirements.returnVersion.reason !== 'succession-or-transfer-of-licence',
+    isFinal: _isFinal(endDate, summer),
+    isSummer: summer,
+    isTwoPartTariff: requirements.twoPartTariff,
+    isUpload: requirements.upload,
+    nald: {
+      regionCode: requirements.returnVersion.licence.region.naldRegionId,
+      areaCode: requirements.returnVersion.licence.areacode,
+      formatId: requirements.legacyId,
+      periodStartDay: requirements.abstractionPeriodStartDay.toString(),
+      periodStartMonth: requirements.abstractionPeriodStartMonth.toString(),
+      periodEndDay: requirements.abstractionPeriodEndDay.toString(),
+      periodEndMonth: requirements.abstractionPeriodEndMonth.toString()
+    },
+    points: _metadataPoints(requirements.returnRequirementPoints),
+    purposes: _metadataPurposes(requirements.returnRequirementPurposes),
+    version: 1
+  }
+}
+
+function _metadataPoints (returnRequirementPoints) {
+  return returnRequirementPoints.map((returnRequirementPoint) => {
+    return {
+      name: returnRequirementPoint.description,
+      ngr1: returnRequirementPoint.ngr1,
+      ngr2: returnRequirementPoint.ngr2,
+      ngr3: returnRequirementPoint.ngr3,
+      ngr4: returnRequirementPoint.ngr4
+    }
+  })
+}
+
+function _metadataPurposes (returnRequirementPurposes) {
+  return returnRequirementPurposes.map((returnRequirementPurpose) => {
+    return {
+      primary: {
+        code: returnRequirementPurpose.primaryPurpose.legacyId,
+        description: returnRequirementPurpose.primaryPurpose.description
+      },
+      secondary: {
+        code: returnRequirementPurpose.secondaryPurpose.legacyId,
+        description: returnRequirementPurpose.secondaryPurpose.description
+      },
+      tertiary: {
+        code: returnRequirementPurpose.purpose.legacyId,
+        description: returnRequirementPurpose.purpose.description
+      }
+    }
+  })
+}
+
+function _id (requirements, startDate, endDate) {
+  const regionCode = requirements.returnVersion.licence.region.naldRegionId
+  const licenceReference = requirements.returnVersion.licence.licenceRef
+  const legacyId = requirements.legacyId
+
+  return `v1:${regionCode}:${licenceReference}:${legacyId}:${startDate}:${endDate}`
+}
+
+function _endDate (summer, returnVersion) {
+  const earliestDate = _earliestDate(summer, returnVersion)
+
+  const _cycleEndDate = cycleEndDate(summer)
+
+  if (earliestDate < _cycleEndDate) {
+    return formatDateObjectToISO(earliestDate)
+  }
+
+  return cycleEndDateAsISO(summer)
+}
+
+function _earliestDate (summer, returnVersion) {
+  const dates = [returnVersion.licence.expiredDate,
+    returnVersion.licence.lapsedDate,
+    returnVersion.licence.revokedDate,
+    returnVersion.endDate]
+    .filter((date) => { return date !== null })
```

Comment:
Returning the value should work in the same way. We also avoid putting these as one-liners. It means we are always consistent, and IMHO, the one-liner fools us into thinking something is more straightforward than it might be.

```suggestion
    .filter((date) => {
      return date
    })
```